third-culture-software/bhima
# Updates for assets scans

Adds scanning of assets from the Assets Registry page.  This PR also fixed a bug that was preventing retrieving the correct asset condition from the last inventory scan.

Closes https://github.com/IMA-WorldHealth/bhima/issues/6719

**TESTING**
- Use bhima_test
- Go to the Assets Management > Assets Registry page
  - Use the Barcode function in the action menu for one of the assets
  - Click on [Menu] > Columns button
    - Select the  **Condition** and "Last Scan Date" options and click the [Submit] button
    -  Click on the [Menu] > Save Grid Configuration
  - Print the barcode or copy the barcode alphanumeric label
  - Click the [+ Scan] button to open a barcode reading window/mode.  Scan the barcode (or paste the copied barcode label).  
    - Update the condition (add a comment if desired)
    - Click [Submit]
  - Notice in the Assets Registry that condition has changed
  - In the Action menu of one of the Assets, click on the **New Asset Scan"" link
  - Update the condition and click [Submit]
  - Notice that the new condition is displayed.
- Go to the Assets Scans Management > Assets Registry page
  - Test the [+ New Asset Scan] button and the corresponding "New Asset Scan" function in an action menu for the asset scan rows. 
  - Notice that new asset scans show on the bottom of the list for each asset. Therefore the last can for an asset contains the current asset condition.  
  - Also, if you edit an asset scan (using the action menu), the condition for that scan will be shown in the edit dialog.  But if you create a new scan, the current condition will be shown.
